### PR TITLE
test_encode: rewrite test cases in criterion

### LINKED ISSUE #2580: convert test_encode to criterion
Use Criterion to rewrite this unit test.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
fixes #2580

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

File: lib/str-repr/tests/test_encode.c [hunk 1]
```diff
@@ -75,21 +53,19 @@ assert_encode_with_forbidden_equals(const gchar *input, const gchar *forbidden_c
   GString *str = g_string_new("");
 
   str_repr_encode(str, input, -1, forbidden_chars);
-  assert_string(str->str, expected, "Encoded value does not match expected");
+  cr_assert_str_eq(str->str, expected, "Encoded value does not match expected");
```
Comment (MrAnno) [inline lib/str-repr/tests/test_encode.c]: The Criterion assertion API is a bit different than the old one. If you specify the assertion description string, the actual and expected values will not be shown.
Could you please add them back?

For example:
```
cr_assert_str_eq(str->str, expected, "Encoded value does not match expected; actual: %s, expected: %s", str->str, expected);
```

Comment (mehul-m-prajapati) [reply lib/str-repr/tests/test_encode.c]: ok. I will change it..

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
@mehul-m-prajapati can you resolve the conflict?

**Comment by mehul-m-prajapati** [comment]
Hi @furiel ,

Ok. I will resolve it.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@mehul-m-prajapati We have some internal issues with @kira-syslogng, we'll fix that ASAP.
Your PR is fine, thanks for the hard work!

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by folti** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by folti** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@folti 🎉

**Pull request merged**